frontend/src/employee-frontend/components/child-information/DailyServiceTimesSection.tsx

```diff
@@ -0,0 +1,706 @@
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import { Loading, Result } from 'lib-common/api'
+import { UUID } from 'lib-common/types'
+import CollapsibleSection from 'lib-components/molecules/CollapsibleSection'
+import { Label, P } from 'lib-components/typography'
+import React, { useContext, useState } from 'react'
+import { useEffect } from 'react'
+import { useTranslation } from '../../state/i18n'
+import { useRestApi } from 'lib-common/utils/useRestApi'
+import { SpinnerSegment } from 'lib-components/atoms/state/Spinner'
+import ErrorSegment from 'lib-components/atoms/state/ErrorSegment'
+import {
+  isRegular,
+  DailyServiceTimes,
+  isIrregular,
+  RegularDailyServiceTimes,
+  IrregularDailyServiceTimes
+} from '../../types/child'
+import {
+  deleteChildDailyServiceTimes,
+  getChildDailyServiceTimes,
+  putChildDailyServiceTimes
+} from '../../api/child/daily-service-times'
+import { faClock, faPen } from '../../../lib-icons'
+import styled from 'styled-components'
+import InlineButton from '../../../lib-components/atoms/buttons/InlineButton'
+import {
+  FixedSpaceColumn,
+  FixedSpaceRow
+} from 'lib-components/layout/flex-helpers'
+import Button from '../../../lib-components/atoms/buttons/Button'
+import { defaultMargins, Gap } from 'lib-components/white-space'
+import Radio from '../../../lib-components/atoms/form/Radio'
+import InputField from '../../../lib-components/atoms/form/InputField'
+import Checkbox from '../../../lib-components/atoms/form/Checkbox'
+import { UIContext } from '../../state/ui'
+import { RequireRole } from '../../utils/roles'
+
+interface Props {
+  id: UUID
+  open: boolean
+}
+
+interface TimeInputRange {
+  start: string
+  end: string
+}
+
+interface SelectableTimeInputRange extends TimeInputRange {
+  selected: boolean
+}
+
+interface FormData {
+  type: 'REGULAR' | 'IRREGULAR' | 'NOT_SET'
+  regular: TimeInputRange
+  monday: SelectableTimeInputRange
+  tuesday: SelectableTimeInputRange
+  wednesday: SelectableTimeInputRange
+  thursday: SelectableTimeInputRange
+  friday: SelectableTimeInputRange
+}
+
+const emptyForm: FormData = {
+  type: 'NOT_SET',
+  regular: {
+    start: '',
+    end: ''
+  },
+  monday: {
+    selected: false,
+    start: '',
+    end: ''
+  },
+  tuesday: {
+    selected: false,
+    start: '',
+    end: ''
+  },
+  wednesday: {
+    selected: false,
+    start: '',
+    end: ''
+  },
+  thursday: {
+    selected: false,
+    start: '',
+    end: ''
+  },
+  friday: {
+    selected: false,
+    start: '',
+    end: ''
+  }
+}
+
+interface ValidationResult {
+  regular: {
+    start: string | null
+    end: string | null
+  }
+  monday: {
+    start: string | null
+    end: string | null
+  }
+  tuesday: {
+    start: string | null
+    end: string | null
+  }
+  wednesday: {
+    start: string | null
+    end: string | null
+  }
+  thursday: {
+    start: string | null
+    end: string | null
+  }
+  friday: {
+    start: string | null
+    end: string | null
+  }
+}
```

Comment:
Samoin tässä tyypin `{ start: string | null, end: string | null }` vois ottaa ulos. Menis ehkä näteimmin aputyypillä `type Nullable<T> = { [K in keyof T]: T[K] | null }` ja sitten `Nullable<TimeInputRange>`.